src/pages/home/report/ReportActionItem.tsx

```diff
@@ -75,201 +75,227 @@ import ReportActionItemMessage from './ReportActionItemMessage';
-    isMostRecentIOUReportAction: PropTypes.bool.isRequired,
+    isMostRecentIOUReportAction: boolean;
 
     /** Should we display the new marker on top of the comment? */
-    shouldDisplayNewMarker: PropTypes.bool.isRequired,
+    shouldDisplayNewMarker: boolean;
 
     /** Determines if the avatar is displayed as a subscript (positioned lower than normal) */
-    shouldShowSubscriptAvatar: PropTypes.bool,
+    shouldShowSubscriptAvatar?: boolean;
 
     /** Position index of the report action in the overall report FlatList view */
-    index: PropTypes.number.isRequired,
-
-    /** Draft message - if this is set the comment is in 'edit' mode */
-    draftMessage: PropTypes.string,
-
-    /** Stores user's preferred skin tone */
-    preferredSkinTone: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
-
-    emojiReactions: EmojiReactionsPropTypes,
-
-    /** IOU report for this action, if any */
-    iouReport: reportPropTypes,
+    index: number;
 
     /** Flag to show, hide the thread divider line */
-    shouldHideThreadDividerLine: PropTypes.bool,
+    shouldHideThreadDividerLine?: boolean;
 
-    /** The user's wallet account */
-    userWallet: userWalletPropTypes,
-
-    /** All the report actions belonging to the report's parent */
-    parentReportActions: PropTypes.objectOf(PropTypes.shape(reportActionPropTypes)),
+    linkedReportActionID?: string;
 
     /** Callback to be called on onPress */
-    onPress: PropTypes.func,
-};
-
-const defaultProps = {
-    draftMessage: undefined,
-    preferredSkinTone: CONST.EMOJI_DEFAULT_SKIN_TONE,
-    emojiReactions: {},
-    shouldShowSubscriptAvatar: false,
-    iouReport: undefined,
-    shouldHideThreadDividerLine: false,
-    userWallet: {},
-    parentReportActions: {},
-    onPress: undefined,
-};
-
-function ReportActionItem(props) {
+    onPress?: () => void;
+} & ReportActionItemOnyxProps;
+
+const isIOUReport = (actionObj: OnyxEntry<OnyxTypes.ReportAction>): actionObj is OnyxTypes.ReportActionBase & OnyxTypes.OriginalMessageIOU =>
+    actionObj?.actionName === CONST.REPORT.ACTIONS.TYPE.IOU;
+
+function ReportActionItem({
+    action,
+    report,
+    linkedReportActionID,
+    displayAsGroup,
+    emojiReactions,
+    index,
+    iouReport,
+    isMostRecentIOUReportAction,
+    parentReportActions,
+    preferredSkinTone = CONST.EMOJI_DEFAULT_SKIN_TONE,
+    shouldDisplayNewMarker,
+    userWallet,
+    shouldHideThreadDividerLine = false,
+    shouldShowSubscriptAvatar = false,
+    policyReportFields,
+    policy,
+    onPress = undefined,
+}: ReportActionItemProps) {
+    const {translate} = useLocalize();
+    const {isSmallScreenWidth} = useWindowDimensions();
+    const blockedFromConcierge = useBlockedFromConcierge();
+    const reportActionDrafts = useReportActionsDrafts();
+    const draftMessage = useMemo(() => getDraftMessage(reportActionDrafts, report.reportID, action), [action, report.reportID, reportActionDrafts]);
```

Comment:
```suggestion
    const draftMessage = useMemo(() => getDraftMessage(reportActionDrafts as NonNullable<OnyxCollection<OnyxTypes.ReportActionsDrafts>>, report.reportID, action), [action, report.reportID, reportActionDrafts]);
```

The return type of `useReportActionsDrafts` is wrong, it probably should be `NonNullable<OnyxCollection<OnyxTypes.ReportActionsDrafts>>` or `OnyxCollection<OnyxTypes.ReportActionsDrafts>`.